src/data-table/column-datetime.js

```diff
@@ -410,7 +410,7 @@ function DatetimeFilter(props) {
                     if (Array.isArray(date)) {
                       if (!date.length) return;
                       const nextDates = date.map((d, i) =>
-                        applyDateToTime(rangeDates[i], d),
+                        d ? applyDateToTime(rangeDates[i], d) : null,
```

Comment:
`d` can now be `undefined` or `null`, sp we need to make sure we don't send a non-date value to `applyDateToTime`.